File: src/main/java/de/florianschmitt/rest/admin/LanguageAdminController.java
package de.florianschmitt.rest.admin;

import org.springframework.data.domain.Sort;
import org.springframework.data.domain.Sort.Direction;
import org.springframework.data.domain.Sort.Order;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import de.florianschmitt.model.entities.ELanguage;
import de.florianschmitt.model.rest.ELanguageAdminDTO;
import de.florianschmitt.service.LanguageService;

@RestController
@RequestMapping("/admin/language")
class LanguageAdminController extends BaseAdminRestController<ELanguage, ELanguageAdminDTO, LanguageService> {

    @Override
    protected Sort getDefaultSortForAll() {
        return new Sort(new Order(Direction.ASC, "viewOrder"));
    }

}

File: src/main/kotlin/de/florianschmitt/rest/admin/LanguageAdminController.kt
package de.florianschmitt.rest.admin

import de.florianschmitt.model.entities.ELanguage
import de.florianschmitt.model.rest.ELanguageAdminDTO
import de.florianschmitt.service.LanguageService
import org.springframework.data.domain.Sort
import org.springframework.data.domain.Sort.Direction
import org.springframework.data.domain.Sort.Order
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/admin/language")
internal class LanguageAdminController : BaseAdminRestController<ELanguage, ELanguageAdminDTO, LanguageService>() {

    override val defaultSortForAll: Sort
        get() = Sort.by(Order(Direction.ASC, "viewOrder"))

}
